partition_version.cc

```diff
@@ -364,6 +369,7 @@ void partition_entry::apply(const schema& s, mutation_partition&& mp, const sche
     }
     new_version->insert_before(*_version);
     set_version(new_version);
+    app_stats.row_insertions += mp.live_row_count(s);
```

Comment:
I think that we still have to count rows from `mp` in case it is necessary to create a new version. Otherwise, if an INSERT operation creates a new version, writes will not be counted.

As for undetected overwrites, I am now accounting for them in `merge_partition_versions`. I managed to produce a situation in which a new version is created and writes are applied to it. The new solution correctly updates overwrite counter when versions are squashed, while the old did not.